### ISSUE TITLE
Autoplaylist playlists not loading iPhone 6S

### ISSUE THREAD

**Issue by Yinka** [simon-weber/Autoplaylists-for-Google-Music#101]
I can't load autoplaylists on my iPhone (6s Plus) after yesterday's IOS Google music app update, am I alone?

**Comment by Yinka** [comment]
Same on my iPad pro (9.7)

**Comment by simon-weber** [comment]
I tested on my android phone (which also just got an update) and everything looks normal. What exactly you mean when you say "can't load"? Are the playlists present but empty? Or maybe not present at all?

**Comment by Yinka** [comment]
The name and description loads fine, with empty playlist. 

Android loads okay here too.

**Comment by simon-weber** [comment]
Hm, that's odd. And all your other playlists work normally? What happens if you create a new autoplaylist -- does that immediately show up as empty?

I have an ios device I can try and reproduce with, but it might be a bit till I can get to it.

**Comment by simon-weber** [comment]
Ok, I tried with an iphone 6: both before and after the update, everything looks normal. On the first launch after the update, it did take about a minute to load my library.

Can you try going into settings and hitting "refresh library" to see if that helps? The mobile clients are stateful; if the initial sync after the update failed for some reason, they might be able to not recover well.

**Comment by Yinka** [comment]
I've refreshed iPhone, iPad and Chrome apps still no joy

When you can, could you please rename one of your playlist to contain an emoji. I have tried multiple times but haven't been able to test that as Google Music keep loading forever and new playlist wont show up.

Just to add everything works okay for me on Android by the way.

Thanks

![image](https://cloud.githubusercontent.com/assets/87303/19412695/2b0e928c-9313-11e6-936a-f1e5be0e1364.png)

**Comment by simon-weber** [comment]
I didn't notice a difference when adding an emoji to the name.

> Google Music keep loading forever

Can you elaborate on this? Is music.google.com failing to load for you?

**Comment by Yinka** [comment]
I have no issue with music.google.com (on chrome) and Android phone.

The issue is with IOS (iPhone and iPad), I have over 70 playlists and only a few non-autoplaylist shared playlists load, and even those few ones are lost the moment I scroll past them, they have to be in focus to load the playlist cover photo which is bizzare

my phone looks like this:

![image](https://cloud.githubusercontent.com/assets/87303/19412843/3f265a22-9316-11e6-8644-580121ef74fc.png)

**Comment by simon-weber** [comment]
Hm. My bet is this is a bug with Google's ios app triggered by many large playlists. Is it possible for you to downgrade to the previous version of the app and see if that fixes it?

**Comment by Yinka** [comment]
I'll file a bug, but not sure how I can downgrade as phone isn't jailbroken

Thanks

**Comment by Yinka** [comment]
I've noticed that most of my playlists have stopped syncing on web too, my "Top Overall" playlist stays at 370 no matter how many times I force sync (by amending the playlist)

Funny though, if I create a new "Top Overall" with same definitions, it syncs ok and I see all 608. so the problem is with existing playlists.

If remove the extension, will I lose all my playlists definitions upon re-install?

**Comment by simon-weber** [comment]
What's your sync period set at? Can you try upping it to something like 720 (12 hours) and see if that helps after a day or two? This helped in a similar situation in https://github.com/simon-weber/Autoplaylists-for-Google-Music/issues/104#issuecomment-257587576.

> If remove the extension, will I lose all my playlists definitions upon re-install?

Yes, unfortunately. You can disable and re-enable the extension without losing anything, but Google removes all your stored data when you uninstall it. I'll consider this another vote for https://github.com/simon-weber/Autoplaylists-for-Google-Music/issues/40 =)

**Comment by Yinka** [comment]
Sync was set to I think every 2 hrs, also I barely opened Chrome for Sync since the issue started anyway.

I'm going to re-create all my playlists and see if that resolves it. it'll probably take me a year to do so :) 

I seem to be the only one with this issue, I wonder why

**Comment by simon-weber** [comment]
Can you try upgrading and [enabling the new syncing apis described here](https://groups.google.com/forum/#!topic/autoplaylists-for-google-music-announcements/fyQ9j-09GNo)? I'm hopeful it'll fix this.

**Comment by Yinka** [comment]
Still no joy, I'm going to switch account anyway, did you mention something about helping to with that?, I remember seeing something similar briefly but cant remember where/when

**Comment by simon-weber** [comment]
Hm, I'm not sure I would have recommended switching accounts: I think that would fix the problem in the short term, but if the account is used in the same way the same problem would happen to it, too.

I'd recommend just disabling the extension while I work on optimizing the new sync. I've laid out all the details here, if you're interested: https://github.com/simon-weber/Autoplaylists-for-Google-Music/issues/97.

**Comment by simon-weber** [comment]
Heads up that I just pushed a big optimization for the new syncing code in 3.2.2. Give it another shot when you have a chance.

**Comment by Yinka** [comment]
Thanks, will do

Y

**Comment by simon-weber** [comment]
Feel free to open a new issue if this is still a problem.

**Comment by Yinka** [comment]
Thanks, I've changed my primary device to Android which didn't have the issue. I haven't noticed the issue re-occur on the iPad though. 

I've also migrated to another google account, reduced my `discover` playlist song limit to 50 (`charting` playlists is still at 1,000 though but I hope to merge couple of them to reduce stress on Big G's server)

Most importantly, finding ways to write more efficient playlist definitions e.g less use of `random`, and using `=` instead of `contain` etc. 

Not sure if those'll help, but better than doing nothing.

Happy new year.
